### PR TITLE
drivers: crc: Add 'final' parameter to crc16()

### PR DESCRIPTION
**Pull request by andrzej-kaczmarek**
This allows to compute CRC16 for data stored in non-contiguous buffers.

Signed-off-by: Andrzej Kaczmarek <andrzej.kaczmarek@codecoup.pl>

### CODE REVIEW

File: drivers/crc/crc16_sw.c
Comment (tbursztyka) [inline drivers/crc/crc16_sw.c]: move it below crc

File: include/crc16.h
Comment (tbursztyka) [inline include/crc16.h]: what about 'pad' as a name? final is too specific to your use-case, imo.

**Review by tbursztyka — CHANGES_REQUESTED** [review]

File: include/crc16.h
Comment (andrzej-kaczmarek) [inline include/crc16.h]: works for me, I'll change it

**Review by tbursztyka — APPROVED** [review]

**Comment by tbursztyka** [comment]
I'd like to get Patrik's approval here before merging, ban I can't add him as reviewer

**Pull request merged**
